source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/admin/settings/organizations/common/form.tsx

```diff
@@ -0,0 +1,287 @@
+import * as React from 'react';
+import { compose } from 'recompose';
+import { Checkbox, Dropdown } from 'semantic-ui-react';
+import { withTemplateHelpers, Mut, Toggle } from 'react-action-decorators';
+
+import { withTranslations, i18nProps } from '@lib/i18n';
+import { isEmpty } from '@lib/collection';
+
+import * as toast from '@lib/toast';
+
+import { OrganizationAttributes } from '@data/models/organization';
+import {
+  query,
+  buildOptions,
+  withLoader,
+  attributesFor,
+  UserResource,
+  OrganizationResource
+} from '@data';
+
+
+interface IOwnProps {
+  organization: OrganizationResource;
+  owner: UserResource;
+  users: UserResource[];
+  onSubmit: (attributes: OrganizationAttributes, relationships: any) => void;
+  onCancel: () => void;
+}
+
+interface IState {
+  name?: string;
+  websiteUrl?: string;
+  owner?: UserResource;
+  ownerError?: string;
+  buildEngineUrl?: string;
+  buildEngineApiAccessToken?: string;
+  logoUrl?: string;
+  publicByDefault?: boolean;
+}
+
+type IProps =
+  & i18nProps
+  & IOwnProps;
+
+
+@withTemplateHelpers
+class OrganizationForm extends React.Component<IProps, IState> {
+  mut: Mut;
+  toggle: Toggle;
+
+  constructor(props: IProps) {
+    super(props);
+
+    const { organization, owner } = props;
+
+    const {
+      name, websiteUrl, logoUrl,
+      buildEngineUrl, buildEngineApiAccessToken,
+      publicByDefault
+    } = attributesFor(organization);
+
+    this.state = {
+      name: name || '',
+      websiteUrl: websiteUrl || '',
+      buildEngineUrl: buildEngineUrl || '',
+      buildEngineApiAccessToken: buildEngineApiAccessToken || '',
+      logoUrl: logoUrl || '',
+      publicByDefault: publicByDefault || false,
+      owner: owner || null,
+      ownerError: ''
+    };
+
+  }
+
+  isValidForm = () => {
+    const { owner } = this.state;
+    const { t } = this.props;
+
+    const ownerError = isEmpty(owner) ? t('admin.settings.organizations.emptyOwner') : '';
+    this.setState({ownerError});
```

Comment:
why set the ownerError even if there is no error?

Reply:
this can be done in two ways.
Clean up the error here or add a validation in the template. I prefer this one (cleaner templates)

Reply:
I just mean that semantically, it's _weird_ that you're setting the error to empty string if the owner isn't empty.
But sure